<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PWM</class>
 <widget class="QWidget" name="PWM">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>716</width>
    <height>238</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QScrollArea" name="help_lay">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>50</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="baseSize">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="help_layContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>774</width>
        <height>85</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <widget class="QLabel" name="Help">
         <property name="font">
          <font>
           <family>Monospace</family>
           <pointsize>9</pointsize>
           <italic>false</italic>
          </font>
         </property>
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-family:'Hack';&quot;&gt;Usage: pwm &amp;lt;frequency (1Hz to 42MHz)&amp;gt; [duty-cycle (0 to 100%)] [exit]&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-family:'Hack';&quot;&gt;help PWM1 (PB11) frequency PWM frequency &amp;lt;value 1Hz to 42MHz&amp;gt; duty-cycle Duty Cycle in % &amp;lt;value 0 to 100&amp;gt;&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-family:'Hack';&quot;&gt;exit Exit PWM mode (reinit PWM1 pin to safe mode/in)&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="widget_2" native="true">
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Start, Hz</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="sPWM_startF">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>20</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1000000</number>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>27</width>
          <height>22</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Stop, Hz</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="sPWM_stopF">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>20</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1000000</number>
        </property>
        <property name="value">
         <number>100000</number>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_5">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>27</width>
          <height>22</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Step, Hz</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="sPWM_FStep">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>20</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>10000</number>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_9">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Expanding</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>27</width>
          <height>22</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Period, ms</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="sPWM_TRamp">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>20</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="minimum">
         <number>40</number>
        </property>
        <property name="maximum">
         <number>10000</number>
        </property>
        <property name="value">
         <number>1000</number>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Expanding</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>27</width>
          <height>22</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="bPWM_Ramp">
        <property name="text">
         <string>Run Ramp</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="autoRepeat">
         <bool>false</bool>
        </property>
        <property name="autoRepeatDelay">
         <number>1000</number>
        </property>
        <property name="autoRepeatInterval">
         <number>1000</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="widget" native="true">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Freq, Hz</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="sFreq">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>20</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1000000</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDial" name="dial_Freq">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1000000</number>
        </property>
        <property name="wrapping">
         <bool>false</bool>
        </property>
        <property name="notchTarget">
         <double>100.000000000000000</double>
        </property>
        <property name="notchesVisible">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>68</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Duty-cycle, %</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="sDirty">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>20</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="maximum">
         <number>100</number>
        </property>
        <property name="value">
         <number>50</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDial" name="dial_Duty">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
        <property name="maximum">
         <number>100</number>
        </property>
        <property name="notchTarget">
         <double>10.000000000000000</double>
        </property>
        <property name="notchesVisible">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>68</width>
          <height>47</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="bdialSend">
        <property name="text">
         <string>LOCK</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="bPWM_Send">
        <property name="text">
         <string>SEND</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>sFreq</sender>
   <signal>valueChanged(int)</signal>
   <receiver>PWM</receiver>
   <slot>guiChange(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>120</x>
     <y>246</y>
    </hint>
    <hint type="destinationlabel">
     <x>454</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sDirty</sender>
   <signal>valueChanged(int)</signal>
   <receiver>PWM</receiver>
   <slot>guiChange(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>492</x>
     <y>246</y>
    </hint>
    <hint type="destinationlabel">
     <x>454</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dial_Freq</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sFreq</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>225</x>
     <y>246</y>
    </hint>
    <hint type="destinationlabel">
     <x>120</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dial_Duty</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sDirty</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>597</x>
     <y>246</y>
    </hint>
    <hint type="destinationlabel">
     <x>492</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bPWM_Send</sender>
   <signal>clicked()</signal>
   <receiver>PWM</receiver>
   <slot>guiSend()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>853</x>
     <y>246</y>
    </hint>
    <hint type="destinationlabel">
     <x>454</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bPWM_Ramp</sender>
   <signal>clicked(bool)</signal>
   <receiver>PWM</receiver>
   <slot>guiRamp(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>853</x>
     <y>149</y>
    </hint>
    <hint type="destinationlabel">
     <x>454</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bdialSend</sender>
   <signal>clicked(bool)</signal>
   <receiver>PWM</receiver>
   <slot>guiDial(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>768</x>
     <y>246</y>
    </hint>
    <hint type="destinationlabel">
     <x>454</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sPWM_TRamp</sender>
   <signal>valueChanged(int)</signal>
   <receiver>PWM</receiver>
   <slot>guiChange(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>718</x>
     <y>149</y>
    </hint>
    <hint type="destinationlabel">
     <x>454</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>guiChange(int)</slot>
  <slot>guiRamp(bool)</slot>
  <slot>guiDial(bool)</slot>
  <slot>guiSend()</slot>
 </slots>
</ui>
